sky/backends/onprem_utils.py

```diff
@@ -181,9 +182,16 @@ def check_and_install_local_env(ips: List[str], auth_config: Dict[str, str]):
     def _install_and_check_dependencies(
             runner: command_runner.SSHCommandRunner) -> None:
         # Checks for python3 installation.
-        backend_utils.run_command_and_handle_ssh_failure(
+        python_version = backend_utils.run_command_and_handle_ssh_failure(
             runner, ('python3 --version'),
```

Comment:
A quick question: since we allow the user to specify the python path in the cluster yaml, does python3 automatically use that python?
```
# Path to the python binary to be used by SkyPilot. Must be the same on all nodes and executable by all users.
python: /usr/bin/python3
```

Reply:
Yes, the `python3` in this code directly refers to the `python` in the cluster yaml. See here: 

https://github.com/skypilot-org/skypilot/blob/fe00a243d0863348c80172f7981f083e954c4992/sky/backends/onprem_utils.py#L458